Lesson3/ConsoleAppExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Project.Interface;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Project.ConsoleApp
+{
+    public static class ConsoleAppExtensions
+    {
+        //argument types
+        public static readonly string Json = "-json";
+        public static readonly string Xml = "-xml";
+
+        /// <summary>
+        /// GetJsonData
+        /// </summary>
+        /// <param name="clArguments">command line arguments</param>
+        /// <returns></returns>
+        public static string[] GetJsonData(this string[] clArguments)
+        {
+            return clArguments.GetFilePathFromArgument(Json).GetFileData(null);
+        }
+
+        /// <summary>
+        /// GetXmlData
+        /// </summary>
+        /// <param name="clArguments">command line arguments</param>
+        /// <returns></returns>
+        public static string[] GetXmlData(this string[] clArguments)
+        {
+            return clArguments.GetFilePathFromArgument(Xml).GetFileData(null);
+        }
+
+        /// <summary>
+        /// GetFileData
+        /// </summary>
+        /// <param name="filePath">file location for data to be parsed</param>
+        /// <param name="ifh">interface for the parser type example JsonParser</param>
+        /// <returns>returns properly parsed file data</returns>
+        private static string[] GetFileData(this string filePath, IFileHandling ifh)
+        {
+            string[] parsedData = null;
+
+            if (ifh != null && !string.IsNullOrWhiteSpace(filePath))
+            {
+                string fileData = File.ReadAllText(filePath);
+                parsedData = ifh.GetParsedData(fileData);
+            }
+
+            return parsedData ?? Array.Empty<string>();
+        }
+
+        /// <summary>
+        /// GetFilePathFromArgument
+        /// </summary>
+        /// <param name="clArguments">command line arguments</param>
+        /// <param name="argumentNameValue">the command line argument type being searched for. Example: -json</param>
+        /// <returns>file path or empty string if can't find that argumentnamevalue</returns>
+        private static string GetFilePathFromArgument(this string[] clArguments, string argumentNameValue)
+        {
+            string filePath = clArguments.SkipWhile(a => string.Compare(a, argumentNameValue, true) != 0)
+                .Skip(1)
+                .DefaultIfEmpty(string.Empty)
```

Comment:
Purposeful Code: Pretty sure this is not needed as FirstOrDefault() will already return an empty string if there is nothing in the collection.